app/plugin.go

```diff
@@ -394,3 +398,48 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 
 	return resp, nil
 }
+
+// notifyPluginEnabled notifies web clients for all cluster peers if
+// manifest version is same across all peers, including self.
+func (a *App) notifyPluginEnabled(manifest *model.Manifest) *model.AppError {
+	if manifest.HasClient() {
+		var statuses model.PluginStatuses
+		expectedPeers := 1
+		actualPeers := 0
+
+		if a.Cluster != nil {
+			var err *model.AppError
+			statuses, err = a.Cluster.GetPluginStatuses()
+			if err != nil {
+				return err
+			}
+			if infos := len(a.Cluster.GetClusterInfos()); infos > 0 {
+				expectedPeers = infos
+			}
+		}
+
+		localStatus, err := a.GetPluginStatus(manifest.Id)
+		if err != nil {
+			return err
+		}
+		statuses = append(statuses, localStatus)
+
+		for _, status := range statuses {
+			// Count consistent peers
+			isPeerConsistent := status.Version == manifest.Version &&
```

Comment:
It seems like this might be unstable. We are checking parameters that are not directly relevant to our goal. All we really want to check is that all active nodes have copied over the file. Maybe we can create a flag just for that? Otherwise if the plugin happens to crash or the versions are incompatible this will break.

Reply:
> All we really want to check is that all active nodes have copied over the file.

copied over the file on to the local FS?

what we really want to check is whether the plugin is enabled or not, as that's when we generate the webapp bundle. It is possible that the plugin is installed (files are copied over on to the local FS) but the webapp bundle isn't extracted, which will result in a 404 if clients reach out to the server asking for the plugin webapp bundle.

Reply:
@crspeller 
but i do agree that this will be flaky, how about we check the webapp bundle hash?